src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -433,15 +473,20 @@ static int _PaPulseAudio_ProcessAudio(PaPulseAudio_Stream *stream,
 
             pulseaudioOutputWritten += pulseaudioOutputBytes;
         }
+        else if( ret != paContinue && isOutputCb && bufferData )
+        {
+            pa_stream_cancel_write( stream->outputStream );
+            bufferData = NULL;
+        }
 
-        hostFrameCount =
-            PaUtil_EndBufferProcessing( &stream->bufferProcessor,
-                                        &ret );
-
-        PaUtil_EndCpuLoadMeasurement( &stream->cpuLoadMeasurer,
-                                      hostFrameCount );
+        if( ret != paContinue )
```

Comment:
This break could be removed and the while loop could be use:

    do
    {
     .....
    } while( ret == paContinue );

Reply:
This was one good idea an I incorporated that